build.gradle

```diff
@@ -70,24 +70,98 @@ allprojects {
                 // skipping Immutable* explicitly due to https://github.com/google/error-prone/issues/1165
                 '-XepExcludedPaths:.*(generated|Immutable).*',
                 '-XepOpt:NullAway:AnnotatedPackages=com.palantir',
-                '-Xep:AmbiguousMethodReference:ERROR',
-                '-Xep:BoxedPrimitiveConstructor:ERROR',
-                '-Xep:CatchFail:ERROR',
-                '-Xep:ClassCanBeStatic:ERROR',
-                '-Xep:DefaultCharset:ERROR',
-                '-Xep:FunctionalInterfaceClash:ERROR',
-                '-Xep:JavaTimeDefaultTimeZone:ERROR',
-                '-Xep:MethodCanBeStatic:ERROR',
-                '-Xep:NullAway:ERROR',
-                '-Xep:NonCanonicalStaticMemberImport:ERROR',
-                '-Xep:PreferSafeLoggableExceptions:ERROR',
-                '-Xep:PreferSafeLoggingPreconditions:ERROR',
-                '-Xep:ReturnMissingNullable:ERROR',
-                '-Xep:Slf4jLogsafeArgs:ERROR',
-                '-Xep:SwitchStatementDefaultCase:ERROR',
-                '-Xep:UnnecessaryParentheses:ERROR',
-                '-Xep:UnusedVariable:ERROR',
             ]
+
+            def warningsToPromote = [
```

Comment:
Should we use the list from errorprone?
https://github.com/google/error-prone/blob/3d225c1eec6d8dada11730c9e9199b710ea26495/core/src/main/java/com/google/errorprone/scanner/BuiltInCheckerSuppliers.java#L593

Reply:
Possibly, will give it a shot adding all non-default enabled. Last time I looked at the non-default ones there were some pathologically slow ones.

Reply:
done